waspc/src/Wasp/Generator/WebAppGenerator.hs

```diff
@@ -74,15 +75,15 @@ genWebApp spec = do
     genFileCopy = return . C.mkTmplFd
 
 genDotEnv :: AppSpec -> Generator [FileDraft]
-genDotEnv spec = return $
-  case AS.dotEnvClientFile spec of
-    Just srcFilePath
-      | not $ AS.isBuild spec ->
-          [ createCopyFileDraft
-              (C.webAppRootDirInProjectRootDir </> dotEnvInWebAppRootDir)
-              srcFilePath
-          ]
-    _ -> []
+-- Don't generate .env if we are building for production, since .env is to be used only for
```

Comment:
What if the user set some variables they might expect to see during the build phase? 

Maybe it's something like `IS_STAGE_BUILD` and they change their layout based on this?

Reply:
Philosophy of dotenv is that it shouldn't have anything to do with build/production, only with development, check it here: https://www.npmjs.com/package/dotenv .

If we figure out we need something like what you mentioned, we can relax this and think about it, but for now we are just sticking to what they say. We also make this clear in docs so there should be no confusion.

Reply:
Understood, makes sense 👍